src/language-js/printer-estree.js

```diff
@@ -4122,12 +4135,9 @@ function printClass(path, options, print) {
 
 function printOptionalToken(path) {
   const node = path.getValue();
-  if (!node.optional) {
-    return "";
-  }
```

Comment:
I guess this should stay for other parsers

Reply:
Yeah there are other types of optional nodes like optional class properties in TypeScript.